src/order/conditionally_complete_lattice.lean

```diff
@@ -692,3 +717,143 @@ noncomputable instance with_top.with_bot.complete_lattice {α : Type*}
   ..with_top.with_bot.bounded_lattice }
 
 end with_top_bot
+
+section subtype
+variables (s : set α)
+
+/-! ### Subtypes of conditionally complete linear orders
+
+In this section we give conditions on a subset of a conditionally complete linear order, to ensure
+that the subtype is itself conditionally complete.
+
+We check that `Ioo` satisfies these conditions.
+
+TODO There are innumerable possible variants -- the interval `Ioo` could be changed to `Ico`, `Icc`,
+`Ioi`, etc.; the `conditionally_complete_linear_order` could be changed to
+`conditionally_complete_linear_order_bot` or `complete_linear_order`.
```

Comment:
Is this section documentation still up-to-date?

Reply:
Thanks, good catch!